scripts/addons.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Elico Corp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import os
+import re
+from urlparse import urlparse
+from subprocess import call
+from subprocess import check_output
+
+EXTRA_ADDONS_PATH = '/mnt/data/additional_addons/'
+ODOO_CONF = '/opt/odoo/etc/odoo.conf'
+ADDONS_PATH = ['/opt/odoo/sources/odoo/addons']
+DEFAULT_SCHEME = 'https://'
+DEFAULT_GIT_REPO_HOSTING_SERVICE = 'github.com'
+DEFAULT_ORGANIZATION = 'OCA'
+
+
+class Repo(object):
+    """
+    oca_dependencies.txt
+
+    For public repo:
+
+    oca-repo -> https://github.com/OCA/oca-repo (branch: default_branch)
+    oca-repo 8.0 -> https://github.com/OCA/oca-repo (branch: 8.0)
+    organization/public-repo -> https://github.com/organization/public-repo (branch: default_branch)
+    organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+    https://github.com/organization/public-repo -> https://github.com/organization/public-repo (branch: default_branch)
+    https://github.com/organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+    public-repo_rename https://github.com/organization/public-repo -> https://github.com/organization/public-repo (branch: default_branch)
+    public-repo_rename https://github.com/organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+
+    For private repo:
+
+    git@github.com:Elico-Corp/private-repo
+    git@github.com:Elico-Corp/private-repo 8.0
+    private_repo_rename git@github.com:Elico-Corp/private-repo
+    private_repo_rename git@github.com:Elico-Corp/private_repo 8.0
+
+    """
+    def __init__(self, dependent, parent=None):
+        if parent:
+            self.parent = parent
+            self.branch = self.parent.branch
+        else:
+            self.parent = None
+            self.branch = None
+        self.dependent = dependent
+        self.folder_name = None
+        self.organization = DEFAULT_ORGANIZATION
+        self.repository = None
+        self.scheme = DEFAULT_SCHEME
+        self.git_repo_host = DEFAULT_GIT_REPO_HOSTING_SERVICE
+        self._parse()
+
+    def _set_branch(self, branch):
+        if branch:
+            self.branch = branch
+        else:
+            self.branch = self.parent.branch
+
+    def _check_is_ssh(self, url):
+        # TODO For other hosting services, this part should be dynamic.
```

Comment:
not a priority but a good point (for example we might have to support bitbucket or other repos in the future)